### PR TITLE
Refactor read_yaml method

### PR DESCRIPTION
**Pull request by liam-ngy**

### CODE REVIEW

File: lib/honitor_bot.rb [hunk 1]
```diff
@@ -48,8 +51,13 @@ def no_changes?
     def read_config
       config = YAML.load_file('config.yml')
 
+      @user_config = UserConfig.new
       config['config'].each do |key, value|
-        instance_variable_set("@#{key}", value)
+        @user_config.name      = value if key == 'name'
```
Comment (nicolas-heinig) [inline lib/honitor_bot.rb]: Try out `send(key + '=', value)`

**Review by nicolas-heinig — COMMENTED** [review]
We can eliminate dependencies when we use [this](https://github.com/jgroeneveld/attributed_object) instead of virtus. What do you think?

File: lib/honitor_bot.rb (hunk 1 above)
Comment (nicolas-heinig) [reply lib/honitor_bot.rb]: Or if it works: `@user_config = UserConfig.new(config['config'])`

**Comment by liam-ngy** [comment]
Closed #12

**Pull request merged**